forge/docker.py

```diff
@@ -15,6 +15,30 @@
 import base64, boto3, os, urllib2, hashlib
 from tasks import task, TaskError, get, sh, Secret
 
+
+class DockerImageBuilderError(Exception):
+
+    report_traceback = False
+    pass
+
+class DockerImageBuilder(object):
+
+    DOCKER = 'docker'
+    IMAGEBUILDER = 'imagebuilder'
+
+    @classmethod
+    def get_cmd_from_name(self, str):
+        if str == self.DOCKER:
+            def docker_build(directory, dockerfile, img, buildargs):
+                return ["docker", "build", directory, "-f", dockerfile, "-t", img] + buildargs
+            return docker_build
+        elif str == self.IMAGEBUILDER:
+            def imagebuilder_build(directory, dockerfile, img, buildargs):
+                return ["imagebuilder", "-f", dockerfile, "-t", img] + buildargs + [directory]
+            return imagebuilder_build
+
+        raise DockerImageBuilderError("No build method named %s exists. Available method names are: %s" % (str, ", ".join(methods)))
```

Comment:
I believe if you make this a TaskError it will be displayed in a prettier way (or if it's not, then that's probably a separate bug).

Reply:
Thanks for the info! Sounds great. I'll address it soon.